crates/anvil-polkadot/src/api_server/server.rs

```diff
@@ -411,7 +419,40 @@ impl ApiServer {
 
 async fn create_revive_rpc_client(substrate_service: &Service) -> Result<EthRpcClient> {
     let rpc_client = RpcClient::new(InMemoryRpcClient(substrate_service.rpc_handlers.clone()));
-    let api = OnlineClient::<SrcChainConfig>::from_rpc_client(rpc_client.clone()).await?;
+
+    // Using best_hash because genesis hash is only set if genesis block number is equal to 0, but
+    // Anvil allows a custom genesis block number.
+    // https://github.com/paritytech/polkadot-sdk/blob/62c3a80f913a272c4f5dba2c91320056d39ec68e/substrate/client/api/src/in_mem.rs#L178
+    let genesis_hash = substrate_service.client.chain_info().best_hash;
```

Comment:
This will probably work in most cases but it's a bit racy.

The genesis hash here may be different if the node is configured to mine blocks very frequently, it could get to mine a block before we retrieve the first block's hash here.

The clean solution would be to retrieve the block hash of the genesis block number (which we know).
Let's do this instead

Reply:
This is why I see tests failing in CI with:
```
called `Result::unwrap()` on an `Err` value: "Expected success but got error: RpcError { code: InternalError, message: \"Client error: RPC error: RPC error: client error: ErrorObject { code: ServerError(1010), message: \\\"Invalid Transaction\\\", data: Some(RawValue(\\\"Transaction call is not expected\\\")) }\", data: None }"
```